frontend/src/views/Waiting.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Message } from 'stompjs';
+import { LargeText, UserNicknameText } from '../components/core/Text';
+import {
+  addUser, SUBSCRIBE_URL, connect,
+  deleteUser, SOCKET_ENDPOINT, subscribe, User,
+} from '../api/Socket';
+import ErrorMessage from '../components/core/Error';
+
+type WaitingRoomPageProps = {
+  nickname: string;
+}
+
+function WaitingRoomPage() {
+  const location = useLocation<WaitingRoomPageProps>();
+  const { nickname } = location.state;
+
+  // Hold error text.
+  const [error, setError] = useState('');
+
+  // Variable to hold the users on the page.
+  const [users, setUsers] = useState<User[]>([]);
+
+  // Variable to hold whether the user is connected to the socket.
+  const [socketConnected, setSocketConnected] = useState(false);
+
+  /**
+   * Nickname that is populated if the join page is on the waiting room stage.
+   * Set error if no nickname is passed in despite the waiting room stage.
+   */
+  if ((!location || !location.state || !location.state.nickname) && !error) {
+    setError('No nickname was provided for the user in the waiting room.');
+  }
```

Comment:
Thoughts on moving this and lines 77-79 into a useEffect hook that runs on page load (and with a [location] [dependency](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects))? That way we might not need the socketConnected variable and can also make sure it runs exactly once after the component mounts.

Reply:
I think that's a great idea -- thanks!